src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPageDefault/EnrollmentAddEventPageDefault.container.js

```diff
@@ -0,0 +1,127 @@
+import { EnrollmentAddEventPageDefaultComponent } from './EnrollmentAddEventPageDefault.component';
+import { deleteEnrollment } from '../../Enrollment/EnrollmentPage.actions';
+import { useWidgetDataFromStore } from '../hooks';
+import {
+    useHideWidgetByRuleLocations,
+} from '../../Enrollment/EnrollmentPageDefault/hooks';
+import { updateEnrollmentEventsWithoutId } from '../../common/EnrollmentOverviewDomain';
+import { dataEntryHasChanges as getDataEntryHasChanges } from '../../../DataEntry/common/dataEntryHasChanges';
+import type { ContainerProps } from './EnrollmentAddEventPageDefault.types';
+
+export const EnrollmentAddEventPageDefault = ({
+    enrollment,
+    attributeValues,
+    commonDataError,
+}: ContainerProps) => {
+    const { programId, stageId, orgUnitId, teiId, enrollmentId } = useLocationQuery();
+
+    const dispatch = useDispatch();
+
+    const handleCancel = useCallback(() => {
+        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    const handleSave = useCallback(
+        (data, uid) => {
+            dispatch(updateEnrollmentEventsWithoutId(uid, data.events[0]));
+            dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
+        },
+        [dispatch, programId, orgUnitId, teiId, enrollmentId],
+    );
+
+    const handleDelete = useCallback(() => {
+        dispatch(batchActions([
+            deleteEnrollment({ enrollmentId }),
+            navigateToEnrollmentPage(programId, orgUnitId, teiId),
+        ]));
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    const widgetReducerName = 'enrollmentEvent-newEvent';
+
+    const dataEntryHasChanges = useSelector(state => getDataEntryHasChanges(state, widgetReducerName));
+    const { program } = useProgramInfo(programId);
+    const selectedProgramStage = [...program.stages.values()].find(item => item.id === stageId);
+    const outputEffects = useWidgetDataFromStore(widgetReducerName);
+    const hideWidgets = useHideWidgetByRuleLocations(program.programRules);
+
+    const rulesExecutionDependencies = useMemo(() => ({
+        events: enrollment?.events,
+        attributeValues,
+        enrollmentData: {
+            enrollmentDate: enrollment?.enrollmentDate,
+            incidentDate: enrollment?.incidentDate,
+            enrollmentId: enrollment?.enrollment,
+        },
+    }), [enrollment, attributeValues]);
+
+    const {
+        handleSetOrgUnitId,
+        handleResetOrgUnitId,
+        handleResetProgramId,
+        handleResetEnrollmentId,
+        handleResetTeiId,
+        handleResetStageId,
+        handleResetEventId,
+        teiDisplayName,
+        trackedEntityName,
+        enrollmentsAsOptions,
+        teiSelectorFailure,
+        userInteractionInProgress,
+    } = useEnrollmentAddEventTopBar(teiId, programId, enrollment);
+
+    if (stageId && !selectedProgramStage) {
```

Comment:
Does it make sense to move the `stageId` validation inside `src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js` near the rest of the ids checks?

Reply:
Yes, I believe that this would be sensible, although this would require us to have the whole program accessible in `EnrollmentAddEventPage.container.js`. To get the program info we would use the `useProgramInfo`-hook, and this is a hook that we try to avoid in the container. (Since it throws an error if programId is missing or invalid, not a controlled message to the user). Let me know if you want me to take a look at any solutions to this. 👍

Reply:
I agree with you, in this case, is best to keep it where it is to avoid the usage of `useProgramInfo` hook in the container. 
Thanks for looking into this!